From 3b8d66a4ced525762c480fcaecc07951cf6fa368 Mon Sep 17 00:00:00 2001
From: Jeffrey Ladouceur <Jeffrey.Ladouceur@freescale.com>
Date: Thu, 15 May 2014 17:37:08 -0400
Subject: [PATCH 463/752] fsl_dce: add error check in test

If there was a error during the process operation the output of zero bytes
was still being allocated. This causes a trace to occur in vmalloc.
Skip allocating and copying output on error.

Signed-off-by: Jeffrey Ladouceur <Jeffrey.Ladouceur@freescale.com>
Change-Id: Ie4f0462902a15aae44a20a7de2478a6488da0e1d
Reviewed-on: http://git.am.freescale.net:8181/12426
Tested-by: Review Code-CDREVIEW <CDREVIEW@freescale.com>
Reviewed-by: Roy Pledge <roy.pledge@freescale.com>
Reviewed-by: Jose Rivera <German.Rivera@freescale.com>
[Original patch taken from QorIQ-SDK-V2.0-20160527-yocto]
Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 drivers/staging/fsl_dce/tests/performance_simple/dce_perf_simple.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/drivers/staging/fsl_dce/tests/performance_simple/dce_perf_simple.c b/drivers/staging/fsl_dce/tests/performance_simple/dce_perf_simple.c
index 39e3348..d6a2cce 100644
--- a/drivers/staging/fsl_dce/tests/performance_simple/dce_perf_simple.c
+++ b/drivers/staging/fsl_dce/tests/performance_simple/dce_perf_simple.c
@@ -940,9 +940,11 @@ done:
 
 	/* unmap the ouput frame */
 	fsl_dce_unmap(qm_fd_addr_get64(&def_process_req->output_fd));
-
-	/* save output */
 	pr_info("Output length is %u\n", def_process_req->dce_cf[0].length);
+
+	/* save output if no error*/
+	if (fsl_dce_get_status(def_process_req->output_fd.status) != STREAM_END)
+		goto skip_output_copy;
 	test_data->out_data = vmalloc(def_process_req->dce_cf[0].length);
 	if (!test_data->out_data)
 		pr_err("Unable to allocate output data\n");
@@ -964,6 +966,7 @@ done:
 		if (ret)
 			pr_err("Error %d\n", __LINE__);
 	}
+skip_output_copy:
 
 	ret = detach_scf64_from_sg(&def_process_req->dce_cf[2],
 			&def_process_req->scf,
-- 
2.9.3

